services/permits/app/permit_conditions/pipelines/CachedAzureOpenAIChatGenerator.py

```diff
@@ -0,0 +1,139 @@
+from app.permit_conditions.pipelines.chat_data import ChatData
+import hashlib
+import struct
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+from haystack.components.generators.chat import AzureOpenAIChatGenerator
+from haystack.dataclasses import ChatMessage
+
+from haystack import component
+import pickle
+import os
+import os
```

Comment:
Imports look a bit strange! import os x 2, and then hashlib & struct are imported again below too.